test/commons/dom/is-focusable.js

```diff
@@ -1,146 +1,657 @@
+
+			assert.isFalse(axe.commons.dom.isNativelyFocusable(el));
+
+		});
+
+		it('should return false for a div with a tabindex with spaces', function () {
+			fixture.innerHTML = '<div id="target" tabindex="0"></div>';
+			var el = document.getElementById('target');
+
+			assert.isFalse(axe.commons.dom.isNativelyFocusable(el));
+
+		});
+
+		it('should return false for a div with a tabindex', function () {
+			fixture.innerHTML = '<div id="target" tabindex="0"></div>';
+			var el = document.getElementById('target');
+
+			assert.isFalse(axe.commons.dom.isNativelyFocusable(el));
 
-		assert.isTrue(axe.commons.dom.isFocusable(el));
+		});
 
+		it('should return false for a div with a non-numeric tabindex', function () {
+			fixture.innerHTML = '<div id="target" tabindex="x"></div>';
+			var el = document.getElementById('target');
+
+			assert.isFalse(axe.commons.dom.isNativelyFocusable(el));
+
+		});
+
+		it('should return true for a details element', function () {
+			fixture.innerHTML = '<details id="target"><p>Detail</p></details>';
+			var el = document.getElementById('target');
+
+			assert.isTrue(axe.commons.dom.isNativelyFocusable(el));
+
+		});
+
+		it('should return false for a div with no tabindex', function () {
+			fixture.innerHTML = '<div id="target"></div>';
+			var el = document.getElementById('target');
+
+			assert.isFalse(axe.commons.dom.isNativelyFocusable(el));
+
+		});
 	});
 
-	it('should return false for a div with no tabindex', function () {
-		fixture.innerHTML = '<div id="target"></div>';
-		var el = document.getElementById('target');
+	describe('dom.insertedIntoFocusOrder', function () {
+
+		var fixture = document.getElementById('fixture');
+
+		beforeEach(function () {
+			fixture.innerHTML = '';
+		});
+
+		it('should return true for span with tabindex 0', function () {
+			fixtureSetup('<span id="spanTabindex0" tabindex="0"></span>');
+			var node = fixture.querySelector('#spanTabindex0');
+
+			assert.isTrue(axe.commons.dom.insertedIntoFocusOrder(node));
+
+		});
+
+		it('should return true for div with tabindex 0', function () {
+			fixtureSetup('<div id="divTabindex0" tabindex="0"></div>');
+			var node = fixture.querySelector('#divTabindex0');
+
+			assert.isTrue(axe.commons.dom.insertedIntoFocusOrder(node));
+
+		});
+
+		it('should return true for heading with tabindex 0', function () {
+			fixtureSetup('<h4 id="headingTabindex0" tabindex="0"></h4>');
+			var node = fixture.querySelector('#headingTabindex0');
+
+			assert.isTrue(axe.commons.dom.insertedIntoFocusOrder(node));
+
+		});
+
+		it('should return true for clipped span with tabindex 0', function () {
+			fixtureSetup('<span id="clippedSpanTabindex0" tabindex="0"></span>');
+			var node = fixture.querySelector('#clippedSpanTabindex0');
+			hideByClipping(node);
+
+			assert.isTrue(axe.commons.dom.insertedIntoFocusOrder(node));
+
+		});
+
+		it('should return true for clipped div with tabindex 0', function () {
```

Comment:
You've added quite a few redundant tests in here. There's no obvious reason why this method would behave different for a `div` or `h4` than it does for a `span`. If you want to keep them, I suggest you combine them. You can have multiple asserts in the same statement. Leaving these tests here implies complexity that isn't there, which can confuse future devs.

Reply:
Removed all but the span tests